planetary-social/nos
# Search relays during mentions autocomplete

## Issues covered
#1560

## Description
This PR includes functionality for autocomplete to also search the query in the user's relay list . It contains the following changes:

- Refactors `AuthorListView` to use `SearchController`.

## How to test
1. Build the code.
2. Tap post button at the bottom of the screen.
3. Tap the @ button or type "@" into the textfield.
4. Type in the search bar.
5. Please check that Authors who weren't in Core Data before the AuthorListView opened are showing up in the list of results.

## Screenshots/Video

Before:

https://github.com/user-attachments/assets/a31797af-91f7-45e7-914a-d9fa8a4bb2dd

After:

https://github.com/user-attachments/assets/a48e355f-84a7-401e-bb73-d3d2ed9acd3d
